### PR TITLE
Update mapping of resource icons & colors

### LINKED ISSUE #6278: Redesign Filetype Icons
# Description

## User Stories

* > As a user I want to visually identify filetypes and folders quickly so that I can find the file I'm looking without effort.

## Value
- lower cognitive costs
- ease of use

## Acceptance Criteria
- Add the following filetype Icons to ODS (maybe to a different location, so that they dont get mixed with our standard icon set):
1. https://fontawesome.com/v5.15/icons/file?style=solid
1. https://fontawesome.com/v5.15/icons/file-alt?style=solid
1. https://fontawesome.com/v5.15/icons/file-archive?style=solid
1. https://fontawesome.com/v5.15/icons/file-audio?style=solid
1. https://fontawesome.com/v5.15/icons/file-code?style=solid
1. https://fontawesome.com/v5.15/icons/file-excel?style=solid
1. https://fontawesome.com/v5.15/icons/file-image?style=solid
1. https://fontawesome.com/v5.15/icons/file-pdf?style=solid
1. https://fontawesome.com/v5.15/icons/file-powerpoint?style=solid
1. https://fontawesome.com/v5.15/icons/file-video?style=solid
1. https://fontawesome.com/v5.15/icons/file-word?style=solid
1. https://fontawesome.com/v5.15/icons/folder?style=solid
- map filetypes and mimetypes/mimetypenamestace "image" "video" to icons and colors
- give appropriate credit (CC-BY-4.0 https://creativecommons.org/licenses/by/4.0/) https://fontawesome.com/license/free

- Let's move the file-extension-type-mapping to the ODS
- Introduce a OcResourceIcon component
- Render the FontAwesome icons smaller in the OcResourceIcon (according to design)
- Introduce (new) themable icon colors in the ODS
- Display FontAwesome icons in separate table in ODS documentation

## Definition of done
- Functional requirements
[ ] functionality described in the user story works
[ ] acceptance criteria are fulfilled
- Quality
[ ] codre review happened
[ ] CI is green
[ ] critical code received unit tests by the developer
[ ] automated tests passed (if automated tests are not available, this test needs to be created and passed
- Non-functional requirements
[ ] no sonar cloud issues

### PR DESCRIPTION
**Pull request by lookacat**
## Description
Adds color mapping to icon representation, needs https://github.com/owncloud/owncloud-design-system/pull/1900 to work

## Related Issue
- Fixes #6278

## Types of changes
- [X] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: packages/web-app-files/src/fileTypeColorMappings.json [hunk 1]
```diff
@@ -0,0 +1,107 @@
+{
+    "7z": "#FFCF01",
+    "accdb": "var(--oc-color-text-default",
+    "ai": "#FF6F00",
+    "apk": "#FFCF01",
+    "bash": "var(--oc-color-text-default",
+    "bz2": "#FFCF01",
+    "c++": "var(--oc-color-text-default",
+    "c": "var(--oc-color-text-default",
+    "cb7": "var(--oc-color-text-default",
```
Comment (pascalwengerter) [inline packages/web-app-files/src/fileTypeColorMappings.json:10]: @lookacat this is where it takes the light/dark colors ;)

File: packages/web-app-files/src/fileTypeColorMappings.json [hunk 2]
```diff
@@ -0,0 +1,107 @@
+{
+    "7z": "#FFCF01",
+    "accdb": "var(--oc-color-text-default",
```
Comment (pascalwengerter) [inline packages/web-app-files/src/fileTypeColorMappings.json:3]:
```suggestion
    "accdb": "var(--oc-color-text-default)",
```
messed it up big time by omitting the closing brackets here -.- someone please fix for the whole file

File: packages/web-app-files/src/fileTypeColorMappings.json (hunk 1 above)
Comment (lookacat) [reply packages/web-app-files/src/fileTypeColorMappings.json]: @pascalwengerter moved it to ODS, its now using only icon color tokens for themes :)

File: packages/web-app-files/src/fileTypeColorMappings.json (hunk 2 above)
Comment (lookacat) [reply packages/web-app-files/src/fileTypeColorMappings.json]: @pascalwengerter moved it to ODS and yes fixed that one ;)

File: packages/web-app-files/src/fileTypeColorMappings.json (hunk 1 above)
Comment (pascalwengerter) [reply packages/web-app-files/src/fileTypeColorMappings.json]: Thx!

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingAccept</strong> https://drone.owncloud.com/owncloud/web/22377/15/1
The following scenarios passed on retry:
- webUISharingAcceptSharesToRoot/acceptShares.feature:108

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingAndUpload</strong> https://drone.owncloud.com/owncloud/web/22377/63/1
The following scenarios passed on retry:
- webUIUpload/upload.feature:94

**Comment by pascalwengerter** [comment]
@kulmann I'd like you to review since I was also involved with this PR

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingInternal3</strong> https://drone.owncloud.com/owncloud/web/22377/59/1
The following scenarios passed on retry:
- webUISharingInternalGroupsSharingIndicator/shareWithGroups.feature:136

**Comment by ownclouders** [comment]
Results for <strong>oCISFiles2</strong> https://drone.owncloud.com/owncloud/web/22377/56/1
The following scenarios passed on retry:
- webUIMarkdownEditor/markdownFile.feature:21

**Review by kulmann — CHANGES_REQUESTED** [review]
Found two small issues, otherwise good to go
- [ ] FileInfo (right sidebar header area) doesn't show filetype icons anymore
- [x] `Document` filetype icon has a really dark blue color in light mode which at least I cannot recognize as blue anymore

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingPublic1</strong> https://drone.owncloud.com/owncloud/web/22395/64/1
The following scenarios passed on retry:
- webUISharingPublicBasic/publicLinkCreate.feature:190

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingPermission</strong> https://drone.owncloud.com/owncloud/web/22395/35/1
The following scenarios passed on retry:
- webUISharingPermissionsUsers/sharePermissionsUsers.feature:121

**Review by kulmann — APPROVED** [review]
LGTM 👍

**Comment by kulmann** [comment]
Previous build was already successful, minor fix that doesn't affect tests. Force merging.

**Pull request merged**